src/compiler-attrs.md

```diff
@@ -0,0 +1,18 @@
+# Compiler attributes
```

Comment:
Once again going to say I don't like the name of this section. I think `crate_name` could be added here, but need to know more about what it actually does to be fully certain. If anything "Compile setting attributes" would be a better name.

Reply:
I'd suggest dropping "compiler" altogether and just have "Recursion limits" as the title until we have more attributes that fit the more broader "settings" title. Otherwise I'd move `type_length_limit` into the type system sections and the other to the macro sections.